src/dataflow/src/sink/kafka.rs

```diff
@@ -574,3 +581,161 @@ where
 
     Box::new(KafkaSinkToken { shutdown_flag })
 }
+
+/// An operator for encoding Rows as Avro that can yield, and doesn't have to process all its in
+/// put in one invocation.
+fn avro_encode_stream<G>(
+    input_stream: &Stream<G, ((Option<Row>, Option<Row>), u64, isize)>,
+    encoder: Encoder,
+    key_schema_id: Option<i32>,
+    value_schema_id: i32,
+    fuel: usize,
+    sink_name: String,
+) -> Stream<G, ((Option<Vec<u8>>, Option<Vec<u8>>), u64, isize)>
+where
+    G: Scope<Timestamp = Timestamp>,
+{
+    let name = format!("{}-avro_encode", sink_name);
+
+    let mut builder = OperatorBuilder::new(name, input_stream.scope());
```

Comment:
so im still working through this change but this seems like a really low level operator building api. is there any reason we can't use `unary_frontier` or `unary` here (I see that we use `OperatorBuilder` elsewhere as well). do we need it to use `build_reschedule` is that the reason?

Reply:
Yes, it is because the operator can yield while still having records in the stash. I thought we then need the more low-level API that allows an operator to signal that it would like to run in the future by a) returning `true` and, b) re-scheduling using the `Activator`.